Updates to fix crashes and add support for Android N

wispborne opened the pull request:
cc @mabaco (https://github.com/mabako) I am including your fixes in here as well because they seem to have fixed the image loading crashes.
--------------------------------------------

Wally is currently broken in newer versions of Android because of android api changes. This PR aims to fix the obvious bugs that were causing crashes.

- Crashes when loading some images (credit to @mabaco)
- Crashes when setting an image as a background on N  (https://inthecheesefactory.com/blog/how-to-share-access-to-file-with-fileprovider-on-android-nougat/en)
- Updating support libraries and breaking changes resulting from there
- Proguard crashes, probably also caused by the support lib updates


You should definitely check out some of the changes that were made by @mabaco, however...there are some Crashlytics changes and other stuff that I didn't look at too closely but may not be desired.

mabako committed:
fixed network threads n stuff

- bumped okhttp version up, which fixes networking on main thread
- bumped other versions, now asks for storage permissions
- removed crashlytics

wispborne committed:
Updated to api 25 and fixed proguard crashes. 

TODO: fix crash when setting desktop background

wispborne committed:
Fixed issues with Android N (couldn't set as background)

wispborne committed:
Hacked in some nicer handling for when the "Access to Files" permission is denied

pmsobrado commented:
Hi, could you provide an APK with the N fixes? Your previous 1.6.0 apk won't install on my phone, although I uninstalled the official one.

Thanks!

davidwhitman commented:
@pmsobrado Just uploaded a new apk. Works fine for me. If it doesn't work for you, let me know what the error is, maybe I'll recognize it.

https://github.com/davidwhitman/wally/releases

pmsobrado commented:
Seems to work great now, thank. Out of curiosity, what was the cause for
crashes on random wallpapers?

Thanks!



* Pablo Manuel Sobrado Gómez*
 Projects area - Devops
 * Emergya Consultoría*
 Tfno: +34 954 51 75 77
 Fax: +34 954 51 64 73
 www.emergya.es


2017-01-23 10:38 GMT+01:00 David Whitman <notifications@github.com>:

> @pmsobrado <https://github.com/pmsobrado> Just uploaded a new apk. Works
> fine for me. If it doesn't work for you, let me know what the error is,
> maybe I'll recognize it.
>
> https://github.com/davidwhitman/wally/releases
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Musenkishi/wally/pull/9#issuecomment-274440191>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AQ4Tm5KKMXW4QsJ3_7RQipTJITiih7LPks5rVHUrgaJpZM4K0a_x>
> .
>

davidwhitman commented:
I don't even remember. I think that those crashes on loading an image were fixed by somebody else, @mabako, and I added a further fix onto his branch.

>edit:  

>>"[bumped okhttp version up, which fixes networking on main thread](https://github.com/davidwhitman/wally/commit/4afb8871574464ed318b36739b6cbd00e33505bf)"

>@mabako updated the version of OkHttp, which fixed `NetworkOnMainThreadException` being thrown and was probably the reason for the crashes. )

What I fixed was mainly being unable to set an image as the desktop wallpaper in N, because of a change in how are you allowed to access files (security change). 

Essentially all I needed to do was change:

`Uri.fromFile(file);` 
to 
`FileProvider.getUriForFile(context, context.getApplicationContext().getPackageName() + ".provider", `

in a couple of places and that's it. 

Oh, and there were some crashes caused by ProGuard (the minification tool) as well. I had to figure out what code it was removing that it shouldn't have been and tell it to stop doing that. And make a small change to how animation is done, because something was deprecated. 

There is a visual bug with some share icons or something now showing when they shouldn't be, but my goal was just to get Wally working again. Would love for @Musenkishi to update the app again, it was ahead of its time.

pmsobrado commented:
Sure haha thanks for the info!



* Pablo Manuel Sobrado Gómez*
 Projects area - Devops
 * Emergya Consultoría*
 Tfno: +34 954 51 75 77
 Fax: +34 954 51 64 73
 www.emergya.es


2017-01-23 11:08 GMT+01:00 David Whitman <notifications@github.com>:

> I don't even remember. I think that those crashes on loading an image were
> fixed by somebody else, @mabako <https://github.com/mabako>, and I added
> a further fix onto his branch.
>
> edit:
>
> "bumped okhttp version up, which fixes networking on main thread
> <https://github.com/davidwhitman/wally/commit/4afb8871574464ed318b36739b6cbd00e33505bf>
> "
>
> @mabako <https://github.com/mabako> updated the version of OkHttp, which
> fixed NetworkOnMainThreadException being thrown and was probably the
> reason for the crashes. )
>
> What I fixed was mainly being unable to set an image as the desktop
> wallpaper in N, because of a change in how are you allowed to access files
> (security change).
>
> Essentially all I needed to do was change:
>
> Uri.fromFile(file);
> to
> FileProvider.getUriForFile(context, context.getApplicationContext().getPackageName()
> + ".provider",
>
> in a couple of places and that's it.
>
> Oh, and there were some crashes caused by ProGuard (the minification tool)
> as well. I had to figure out what code it was removing that it shouldn't
> have been and tell it to stop doing that. And make a small change to how
> animation is done, because something was deprecated.
>
> There is a visual bug with some share icons or something now showing when
> they shouldn't be, but my goal was just to get Wally working again. Would
> love for @Musenkishi <https://github.com/Musenkishi> to update the app
> again, it was ahead of its time.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Musenkishi/wally/pull/9#issuecomment-274446710>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AQ4TmyvcAjt7s42qA9glmUG3MUWnZaXGks5rVHwogaJpZM4K0a_x>
> .
>

wispborne committed:
Changed url from alpha.wallhaven to just wallhaven.

Updated to newest target android API and fixed issues with implicit intents by switching to LocalBroadcastManager. Updated support libs to AndroidX, Updated Glide to latest.

wispborne committed:
New signing key, uploading it this time so I don't lose it.

Setting version to 1.6.1 and fixing obfuscator rules.
